javascript/.gitignore

```diff
@@ -131,4 +131,5 @@ temp/
 # End of https://www.toptal.com/developers/gitignore/api/node
 
 # Custom
-lib/
\ No newline at end of file
+lib/
+package-lock.json
```

Comment:
After discussion, we came to the conclusion that server code *should* include `package-lock.json` and that libraries *should not* include `package-lock.json`.

This change is appropriate.